### PR TITLE
Add support for managed EIP healthchecking

### PR DESCRIPTION
**Pull request by falzm**
This change adds support for the upcoming Exoscale managed Elastic IP feature.

### CODE REVIEW

**Comment by falzm** [comment]
Changes testing:

```console
$ env TF_ACC=1 EXOSCALE_API_KEY=********* EXOSCALE_API_SECRET=*********  go test -mod vendor -v -timeout 30s github.com/terraform-providers/terraform-provider-exoscale/exoscale -run '^(TestAccElasticIP)$'
=== RUN   TestAccElasticIP
--- PASS: TestAccElasticIP (10.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-exoscale/exoscale	10.167s
```

**Comment by mcorbin** [comment]
There is an error when healthcheck timeout/interval is not specified (in that case, it should work because we have default values):

```

resource "exoscale_ipaddress" "ingress" {
  zone = "ch-dk-2"
  healthcheck_mode = "http"
  healthcheck_port = 9000
  healthcheck_path = "/"
}

```
```
exoscale_ipaddress.ingress: Creating...
  healthcheck_mode: "" => "http"
  healthcheck_path: "" => "/"
  healthcheck_port: "" => "9000"
  ip_address:       "" => "<computed>"
  tags.%:           "" => "<computed>"
  zone:             "" => "ch-dk-2"

Error: Error applying plan:

1 error(s) occurred:

* exoscale_ipaddress.ingress: 1 error(s) occurred:

* exoscale_ipaddress.ingress: healthcheck_timeout must be lower than healthcheck_interval
```

**Comment by falzm** [comment]
My bad, didn't test this case. This is now fixed, thanks @mcorbin 👌

**Comment by mcorbin** [comment]
Another issue: If I rul multiple time `terraform apply`, terraoform tries to update the IP all the time (even if I don't uodate the IP in the terraform file):

```
resource "exoscale_ipaddress" "ingress" {
  zone = "ch-dk-2"
  healthcheck_mode = "http"
  healthcheck_port = 9000
  healthcheck_path = "/"
}
```

```
terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + exoscale_ipaddress.ingress
      id:               <computed>
      healthcheck_mode: "http"
      healthcheck_path: "/"
      healthcheck_port: "9000"
      ip_address:       <computed>
      tags.%:           <computed>
      zone:             "ch-dk-2"


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

exoscale_ipaddress.ingress: Creating...
  healthcheck_mode: "" => "http"
  healthcheck_path: "" => "/"
  healthcheck_port: "" => "9000"
  ip_address:       "" => "<computed>"
  tags.%:           "" => "<computed>"
  zone:             "" => "ch-dk-2"
exoscale_ipaddress.ingress: Creation complete after 2s (ID: 3795ff45-1359-4ab8-baa7-a0c97270a27f)
```

```
terraform apply
exoscale_ipaddress.ingress: Refreshing state... (ID: 3795ff45-1359-4ab8-baa7-a0c97270a27f)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ exoscale_ipaddress.ingress
      healthcheck_interval:     "10" => "0"
      healthcheck_strikes_fail: "3" => "0"
      healthcheck_strikes_ok:   "2" => "0"
      healthcheck_timeout:      "2" => "0"


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

exoscale_ipaddress.ingress: Modifying... (ID: 3795ff45-1359-4ab8-baa7-a0c97270a27f)
  healthcheck_interval:     "10" => "0"
  healthcheck_strikes_fail: "3" => "0"
  healthcheck_strikes_ok:   "2" => "0"
  healthcheck_timeout:      "2" => "0"
exoscale_ipaddress.ingress: Modifications complete after 3s (ID: 3795ff45-1359-4ab8-baa7-a0c97270a27f)
```

**Comment by falzm** [comment]
@mcorbin this should now be fixed also. In order to simplify the logic I had to make all attributes mandatory (if `healthcheck_mode` is specified).

File: exoscale/resource_ipaddress.go [hunk 1]
```diff
@@ -5,22 +5,66 @@ import (
...
-			Computed: true,
-		},
 		"zone": {
 			Type:        schema.TypeString,
 			Required:    true,
 			ForceNew:    true,
-			Description: "Name of the Data-Center",
+			Description: "Name of the zone",
+		},
+		"healthcheck_mode": {
```
Comment (pierre-emmanuelJ) [inline exoscale/resource_ipaddress.go]: maybe all these string value in const can be better to be able to reuse it in all this src code

**Comment by pierre-emmanuelJ** [comment]
Nice work 👍 that's make lot's of `if` statement 😄

File: exoscale/resource_ipaddress.go [hunk 2]
```diff
@@ -5,22 +5,66 @@ import (
...
-		"ip_address": {
-			Type:     schema.TypeString,
-			Computed: true,
-		},
 		"zone": {
 			Type:        schema.TypeString,
 			Required:    true,
 			ForceNew:    true,
-			Description: "Name of the Data-Center",
+			Description: "Name of the zone",
```
Comment (pierre-emmanuelJ) [inline exoscale/resource_ipaddress.go]: 👍

File: exoscale/resource_ipaddress.go (hunk 1 above)
Comment (falzm) [reply exoscale/resource_ipaddress.go]: True, but this kind of refactoring will be for another day 😉

**Pull request merged**